packages/rocketchat-api/server/routes.coffee

```diff
@@ -185,5 +196,195 @@ RocketChat.API.v1.addRoute 'groups.create', authRequired: true,
 			return RocketChat.API.v1.failure e.name + ': ' + e.message
 
 		return RocketChat.API.v1.success
-			group: RocketChat.models.Rooms.findOne({_id: id.rid})
+			group: RocketChat.models.Rooms.findOne({_id: id.rid})		
+
+
+#list All Private Groups		
+RocketChat.API.v1.addRoute 'privateRooms.list', authRequired: true,
+	get: ->
+		try
+			this.response.setTimeout (1000)
+			rooms = RocketChat.models.Rooms.findByType('p', { sort: { msgs:-1 } }).fetch()
+			#status: 'success', rooms: rooms
+		catch e
+			return RocketChat.API.v1.failure e.name + ': ' + e.message
+		return RocketChat.API.v1.success
+			list: rooms
+			
+			
+#list All Direct Groups		
+RocketChat.API.v1.addRoute 'directRoom.list', authRequired: true,
+	get: ->
+		try
+			this.response.setTimeout (1000)
+			rooms = RocketChat.models.Rooms.findByType('d', { sort: { msgs:-1 } }).fetch()
+		catch e
+			return RocketChat.API.v1.failure e.name + ': ' + e.message
+		return RocketChat.API.v1.success
+			list: rooms			
+			
+			
+# addUser to a channel/private group
+RocketChat.API.v1.addRoute 'addUser', authRequired: true,
+	post: ->
+		if RocketChat.authz.hasPermission(@userId, 'bulk-create-c')
+			try
+				this.response.setTimeout (1000 * @userId.length)
+				Meteor.runAsUser this.userId, () =>
+					(Meteor.call 'addUserToRoom', rid:@bodyParams.room, username:@bodyParams.username)
+				status: 'success', rid:@bodyParams.room, username:@bodyParams.username
+			catch e
+				return RocketChat.API.v1.failure e.name + ': ' + e.message
+		else
+			console.log '[restapi] addUserToRoom -> '.red, "User does not have 'bulk-create-c' permission"
+			return RocketChat.API.v1.unauthorized e.name + ': ' + e.message
+
+		
+### Remove rooms
+any type i.e. channels/private/direct message room
+user must also have create-c permission
+pass room _id in body of data (  data='{"name":[" room _id"]}'  )				
+###
+RocketChat.API.v1.addRoute 'bulk/removeGroup', authRequired: true,
+	delete: ->
+		if RocketChat.authz.hasPermission(@userId, 'bulk-create-c')
```

Comment:
We should have a permission for delete and not use the `bulk-create-c` permission